cmd/earth/main.go

```diff
@@ -188,7 +191,10 @@ func main() {
 	app := newEarthApp(ctx, conslogging.Current(colorMode, padding))
 	app.autoComplete()
 
-	os.Exit(app.run(ctx, os.Args))
+	startTime := time.Now()
+	exitCode := app.run(ctx, os.Args)
+	app.collectAnalytics(exitCode, time.Since(startTime))
+	os.Exit(exitCode)
```

Comment:
I would also make sure that the user is running `build` here and not some other command.

Reply:
... or perhaps the command itself can be a "property".

Reply:
I added a command property; I couldn't figure out a great way to do it, so I had to make use of the `After` function to record which command was run, but I can't put the call to collectAnalytics in `After` since the returned error wasn't available there.

Reply:
I wonder if `Command.Name` ends up being just `get` in the case of `earth secrets get`? In any case, I think it's a good start regardless.

Reply:
I was wrong with this initial idea, it's just showing up as an empty string, I'll need to figure out another way to get to it. I initially thought including all of argv would work, but it could capture some sensitive user information so that's not an option.

Reply:
A simple idea could be to set the command name ourselves at the beginning of each `action*` fun

Reply:
I'll try that